beets/dbcore/queryparse.py

```diff
@@ -96,17 +96,33 @@ def construct_query_part(model_cls, prefixes, query_part):
     key, pattern, query_class = \
         parse_query_part(query_part, query_classes, prefixes)
 
+    # Handle negation: set negation flag, and recover the original query_class.
```

Comment:
`construct_query_part` has been modified: for every `return` on the original version, if the negation prefix is specified, a negated query `NotQuery(q)` is returned (otherwise, it behaves like the original version, returning `q`).

The original `query_class` is recovered, as it is "swallowed" by `parse_query_part()`. The reason for defaulting to `query.SubstringQuery` if the key was not found is because it seems to be the sane default (used on the signature of  `parse_query_part()` as well) - I hope this is safe enough, but would welcome confirmation.

If everything looks good, the last `else` and `return` (122-128) should probably also be merged, for readability, as it follows the same pattern as the other return cases - it was an oversight on my end.